sdk/keyvault/keyvault-certificates/test/internal/lroUnexpectedErrors.spec.ts

```diff
@@ -0,0 +1,198 @@
+        async getDeletedCertificate(): Promise<any> {
+          const error = new Error(`${code}`);
+          (error as any).statusCode = code;
+          throw error;
+        }
+      };
+      const poller = new DeleteCertificatePoller({
+        vaultUrl,
+        certificateName: "name",
+        client
+      });
+
+      await poller.poll();
+      await poller.poll();
+
+      assert.ok(!poller.getOperationState().isCompleted);
+    });
+
+    it("Errors other than 403 and 404 throw", async function() {
+      const codes = [401, 402, 405, 500];
+      for (const code in codes) {
+        const client: any = {
+          async deleteCertificate(): Promise<any> {
+            return {
+              id: "/version/name/version",
+              recoveryId: "something"
+            };
+          },
+          async getDeletedCertificate(): Promise<any> {
+            const error = new Error(`${code}`);
+            (error as any).statusCode = code;
+            throw error;
+          }
+        };
+        const poller = new DeleteCertificatePoller({
+          vaultUrl,
+          certificateName: "name",
+          client
+        });
+
+        let error: Error | null = null;
+        try {
+          await poller.pollUntilDone();
+        } catch (e) {
+          error = e;
+        }
+
+        assert.equal((error as any).statusCode, code);
+      }
+    });
+  });
+
+  describe("recover LRO", () => {
+    it("403 doesn't throw", async function() {
+      const code = 403;
+      const client: any = {
+        async recoverDeletedCertificate(): Promise<any> {
+          return {
+            _response: {
+              parsedBody: {
+                id: "/version/name/version",
+                recoveryId: "something"
+              }
+            }
+          };
+        },
+        async getCertificate(): Promise<any> {
+          const error = new Error(`${code}`);
+          (error as any).statusCode = code;
+          throw error;
+        }
+      };
+      const poller = new RecoverDeletedCertificatePoller({
+        vaultUrl,
+        certificateName: "name",
+        client
+      });
+
+      await poller.pollUntilDone();
+
+      assert.ok(poller.getOperationState().isCompleted);
+    });
+
+    it("404 doesn't throw", async function() {
+      const code = 404;
+      const client: any = {
+        async recoverDeletedCertificate(): Promise<any> {
+          return {
+            _response: {
+              parsedBody: {
+                id: "/version/name/version",
+                recoveryId: "something"
+              }
+            }
+          };
+        },
+        async getCertificate(): Promise<any> {
+          const error = new Error(`${code}`);
+          (error as any).statusCode = code;
+          throw error;
+        }
+      };
+      const poller = new RecoverDeletedCertificatePoller({
+        vaultUrl,
+        certificateName: "name",
+        client
+      });
+
+      await poller.poll();
```

Comment:
Curious why do we need to poll twice here? Is it a test setup issue or product code issue? Reading it - maybe throwing from the `recoverDeletedCertificate` function will allow us to avoid polling twice?

Reply:
The double poll is here to make sure this behavior is accounted for:

```ts
    if (!state.isStarted) {
      try {
        state.result = await this.getCertificate(certificateName, this.operationOptions);
        state.isCompleted = true;
      } catch (e) {
        // getCertificate will only work once the LRO is completed.
      }
      if (!state.isCompleted) {
        state.result = await this.recoverDeletedCertificate(certificateName, this.operationOptions);
        state.isStarted = true;
      }
    }
```

Reply:
That code is pretty much the same in all of the recover poller operations, we first check if the thing has already been recovered